### PR TITLE
Windows ci

### PR DESCRIPTION
**Pull request by akrieger**
#### Summary
Build "Windows Github CI for Cataclysm and ObjectCreator"

#### Purpose of change

Currently our windows builds are cross compiled from linux with a gcc 5.5 based mingw toolchain. This causes problems because the rest of our supported compiler matrix supports basic c++14 features like constexpr quite well, and being stuck on gcc 5.5 is causing problems. Having native Windows CI for Windows removes this discrepancy and will down the line unblock using c++17 as well.

#### Describe the solution

The CI jobs are based off @jbytheway's initial stabs in https://github.com/CleverRaven/Cataclysm-DDA/pull/49132 which I forked and played with. The key thing making this viable is using the `run-vcpkg` action to cache the build of all the vcpkg provided dependencies. These can take quite a while, especially for ObjectCreator (on the order of 40 minutes, compared to but in addition to 10 minutes for Cataclysm's deps). Having a functional cache puts these builds in the realm of feasible for useful signal. Otherwise as best as I can tell it's just a bog standard Windows build.

#### Testing

Ran these jobs in my fork to test if cache seeding and reuse worked. https://github.com/akrieger/Cataclysm-DDA/actions/runs/1234412229 ran with a cold cache and built ObjectCreator then filled cache. https://github.com/akrieger/Cataclysm-DDA/runs/3602010591 then hit the `run-vcpkg` cache and went straight to building ObjectCreator, no time spent in vcpkg deps. https://github.com/akrieger/Cataclysm-DDA/runs/3602010542 built Cataclysm and has a separate cache so it doesn't waste time or space downloading Qt5 unnecessarily.

#### Additional Context

I also cleaned up the vcpkg usage docs because the out-of-line install is not required. MSBuild will automatically invoke vcpkg when needed so people can just open Visual Studio and run build.

### CODE REVIEW

File: .github/workflows/msvc-full-features.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    - name: Integrate vcpkg
+      run: |
+        vcpkg integrate install
+
+    - name: Build
+      run: |
+          cd msvc-full-features
+          msbuild -m -p:Configuration=Release -p:Platform=x64 -target:Cataclysm-vcpkg-static Cataclysm-vcpkg-static.sln
+
+    - name: Dump vcpkg logs if vcpkg build failed
```
Comment (jbytheway) [inline .github/workflows/msvc-full-features.yml:74]:
```suggestion
    - name: Dump vcpkg logs if build failed
```

File: .github/workflows/object_creator.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    - name: Integrate vcpkg
+      run: |
+        vcpkg integrate install
+
+    - name: Build
+      run: |
+          cd msvc-object_creator
+          msbuild -m -p:Configuration=Release -p:Platform=x64 ObjectCreator-vcpkg-static.sln
+
+    - name: Dump vcpkg logs if vcpkg build failed
```
Comment (jbytheway) [inline .github/workflows/object_creator.yml:72]:
```suggestion
    - name: Dump vcpkg logs if build failed
```

**Review by jbytheway — COMMENTED** [review]
Looks good.

**Comment by ZhilkinSerg** [comment]
We don't need 0.F-dev branch probably

**Comment by akrieger** [comment]
- Merged review suggestions into force push.
- Removed 0.F branch builds.
- Tweaked hashFiles call to actually hash vcpkg.json files because I forgot it requires relative paths.

**Comment by akrieger** [comment]
New builds are looking green. We could potentially run all the builds in the same job to save time by not recompiling cata lib just for ObjectCreator, but to do that we'd have to add ObjectCreator to the main Cata solution. Not the end of the world, mind you, just becomes fiddlier cause the whole solution won't build in debug anymore until the debug error gets resolved. As-is you can't build cata from one solution and use it from a different solution even if it's the same project both times.

**Pull request merged**